update/sync/syncTransactions_test.go

```diff
@@ -341,3 +345,585 @@ func TestSyncPendingTransactionsFor_ReceiveMissingTx(t *testing.T) {
+
+		args := createMockArgs()
+		args.DataPools = &dataRetrieverMock.PoolsHolderStub{
+			ValidatorsInfoCalled: func() dataRetriever.ShardedDataCacherNotifier {
+				return &testscommon.ShardedDataStub{
+					ShardDataStoreCalled: func(cacheID string) storage.Cacher {
+						return &testscommon.CacherStub{
+							PeekCalled: func(key []byte) (value interface{}, ok bool) {
+								if bytes.Equal(key, txHash) {
+									return svi, true
+								}
+								return nil, false
+							},
+						}
+					},
+				}
+			},
+		}
+		transactionsSyncer, _ := NewTransactionsSyncer(args)
+
+		miniBlock := &block.MiniBlock{
+			TxHashes: [][]byte{
+				[]byte("a"),
+				[]byte("b"),
+				[]byte("c"),
+			},
+		}
+		transactionsSyncer.mutPendingTx.Lock()
+		transactionsSyncer.mapTxsToMiniBlocks[string(txHash)] = miniBlock
+		transactionsSyncer.mutPendingTx.Unlock()
+
+		shardValidatorInfo, bFound := transactionsSyncer.getValidatorInfoFromPool(txHash)
+		assert.Equal(t, svi, shardValidatorInfo)
+		assert.True(t, bFound)
+	})
+}
+
+func TestTransactionsSync_GetValidatorInfoFromPoolWithSearchFirstShouldWork(t *testing.T) {
+	t.Parallel()
+
+	txHash := []byte("hash")
+	svi := &state.ShardValidatorInfo{
+		PublicKey: []byte("x"),
+	}
+
+	args := createMockArgs()
+	transactionsSyncer, _ := NewTransactionsSyncer(args)
+
+	// txHash is not found in validatorInfoPool
+	validatorsInfoPool := &testscommon.ShardedDataStub{
+		SearchFirstDataCalled: func(key []byte) (value interface{}, ok bool) {
+			return nil, false
+		},
+	}
+	shardValidatorInfo, bFound := transactionsSyncer.getValidatorInfoFromPoolWithSearchFirst(txHash, validatorsInfoPool)
+	assert.Nil(t, shardValidatorInfo)
+	assert.False(t, bFound)
+
+	// value received from validatorInfoPool is not of type *state.ShardValidatorInfo
+	validatorsInfoPool = &testscommon.ShardedDataStub{
+		SearchFirstDataCalled: func(key []byte) (value interface{}, ok bool) {
+			return nil, true
+		},
+	}
+	shardValidatorInfo, bFound = transactionsSyncer.getValidatorInfoFromPoolWithSearchFirst(txHash, validatorsInfoPool)
+	assert.Nil(t, shardValidatorInfo)
+	assert.False(t, bFound)
+
+	// get validator info from pool with search first should work
+	validatorsInfoPool = &testscommon.ShardedDataStub{
+		SearchFirstDataCalled: func(key []byte) (value interface{}, ok bool) {
+			if bytes.Equal(key, txHash) {
+				return svi, true
+			}
+			return nil, false
+		},
+	}
+	shardValidatorInfo, bFound = transactionsSyncer.getValidatorInfoFromPoolWithSearchFirst(txHash, validatorsInfoPool)
+	assert.Equal(t, svi, shardValidatorInfo)
+	assert.True(t, bFound)
+}
+
+func TestTransactionsSync_GetValidatorInfoFromPoolOrStorage(t *testing.T) {
+	t.Parallel()
+
+	t.Run("get validator info from pool or storage should work from pool", func(t *testing.T) {
+		t.Parallel()
+
+		txHash := []byte("hash")
+		svi := &state.ShardValidatorInfo{
+			PublicKey: []byte("x"),
+		}
+
+		args := createMockArgs()
+		args.DataPools = &dataRetrieverMock.PoolsHolderStub{
```

Comment:
maybe create a function to avoid this duplicated code?

Reply:
Done